warehouse/integrations/snowflake/snowflake_test.go

```diff
@@ -287,43 +295,56 @@ func TestIntegration(t *testing.T) {
 				asyncJob:              true,
 				stagingFilePrefix:     "testdata/sources-job",
 			},
+			{
+				name:                          "Upload Job in append mode",
+				writeKey:                      writeKey,
+				schema:                        namespace,
+				tables:                        []string{"identifies", "users", "tracks"},
+				sourceID:                      sourceID,
+				destinationID:                 destinationID,
+				cred:                          credentials,
+				database:                      database,
+				stagingFilesEventsMap:         testhelper.EventsCountMap{"wh_staging_files": 3},
+				stagingFilesModifiedEventsMap: testhelper.EventsCountMap{"wh_staging_files": 3},
+				loadFilesEventsMap:            map[string]int{"identifies": 1, "users": 1, "tracks": 1},
+				tableUploadsEventsMap:         map[string]int{"identifies": 1, "users": 1, "tracks": 1},
+				warehouseEventsMap:            map[string]int{"identifies": 1, "users": 1, "tracks": 1},
+				warehouseEventsMap2:           map[string]int{"identifies": 2, "users": 1, "tracks": 2},
+				stagingFilePrefix:             "testdata/append-job",
+				appendMode:                    true,
+				customUserID:                  testhelper.GetUserId("append_test"),
+			},
 		}
 
 		for _, tc := range testcase {
 			tc := tc
-
 			t.Run(tc.name, func(t *testing.T) {
-				t.Parallel()
-
-				cred := tc.cred
-				cred.Database = tc.database
-
-				urlConfig := snowflakedb.Config{
-					Account:   cred.Account,
-					User:      cred.User,
-					Role:      cred.Role,
-					Password:  cred.Password,
-					Database:  cred.Database,
-					Warehouse: cred.Warehouse,
+				cancel := bootstrap(tc.appendMode)
+				defer cancel()
```

Comment:
Should we wait here for a graceful shutdown of the server by utilizing either the svcDone channel or passing testing.T of the subtest in bootstrap?

Reply:
Currently `t.Cleanup(func() { <-svcDone })` is being registered at the `TestIntegration` level.

Reply:
@achettyiitr good catch, addressed [here](https://github.com/rudderlabs/rudder-server/pull/3745/commits/2be8bfc54e7d4c1f00c863dc3e293ee2f57abf6a).